parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -383,10 +383,7 @@ public DeleteWriteBuilder equalityFieldIds(int... fieldIds) {
 
       meta("delete-type", "position");
 
-      if (rowSchema != null) {
-        Preconditions.checkState(createWriterFunc != null,
-            "Cannot create delete file with deletes rows unless createWriterFunc is set");
-
+      if (rowSchema != null && createWriterFunc != null) {
```

Comment:
Checking for `createWriterFunc` here is needed because `forTable` sets the row schema. So rows can be included in position deletes when the writer func is added and when there is a row schema.